app/src/main/java/com/akp/ceg4110/quickreports/DatabaseAccessor.java

```diff
@@ -0,0 +1,234 @@
+package com.akp.ceg4110.quickreports;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DatabaseAccessor
+{
+
+    public static final String DATABASE_NAME = "Incidents";
+
+    public static final String INCIDENT_TABLE = "incident_table";
+    public static final String NAME_COLUMN = "name";
+    public static final String DESCRIPTION_COLUMN = "description";
+    public static final String INCIDENT_WEATHER = "weather";
+
+    public static final String PICTURE_TABLE  = "image_table";
+    public static final String PICTURE_NAME_COLUMN = "name";
+    public static final String PICTURE_PRIMARY_COLUMN = "picture_reference";
+    public static final String PICTURE_COLUMN = "picture";
+
+    private SQLiteDatabase db;
+
+
+    public DatabaseAccessor(SQLiteDatabase db){
+        this.db = db;
+        //Build string for creating the incident_table table
+        //TEMPLATE:
+        //CREATE TABLE IF NOT EXISTS incident_table (name VARCHAR(255), description VARCHAR(255), PRIMARY KEY (name));
+        String createIncidentTable = String.format("CREATE TABLE IF NOT EXISTS %1$s (%2$s VARCHAR(255), %3$s VARCHAR(255), %4$s VARCHAR(500), PRIMARY KEY(%5$s));",
+                INCIDENT_TABLE, NAME_COLUMN, DESCRIPTION_COLUMN, INCIDENT_WEATHER, NAME_COLUMN);
+        //Build string for creating the image_table table
+        //TEMPLATE:
+        //CREATE TABLE IF NOT EXISTS image_table (picture_reference INTEGER PRIMARY KEY AUTOINCREMENT, name VARCHAR(255), picture BLOB);
+        String createPictureTable = String.format("CREATE TABLE IF NOT EXISTS %1$s ( %2$s INTEGER PRIMARY KEY AUTOINCREMENT, %3$s VARCHAR(255), %4$s BLOB);",
+                PICTURE_TABLE, PICTURE_PRIMARY_COLUMN, PICTURE_NAME_COLUMN, PICTURE_COLUMN, PICTURE_PRIMARY_COLUMN);
+        try{
+            db.execSQL(createIncidentTable); //Add incident table to DB
+            db.execSQL(createPictureTable);   //Add picture table to DB
+        }catch(Exception e){
+            System.out.println("Error creating table: " + e.getMessage());
+        }
+    }
+
+    public void addIncident(Incident incident){
+        //TEMPLATE:
+        //INSERT INTO incident_table VALUES ('incident.getName()', 'incident.getDescription()');
+        //Also written as:
+        //INSERT INTO incident_table VALUES (incident.toString());
+        //Doesn't include the images
+        String insertIncidentTable = String.format("INSERT INTO %1$s VALUES ('%2$s', '%3$s', '%4$s');",
+                INCIDENT_TABLE, incident.getName(), incident.getDescription(), incident.getWeather());
+        List<Bitmap> images = incident.getImages();
```

Comment:
I'm confused about this line. If we're adding an incident for the first time, there shouldn't be any images to "get" right?

Reply:
I am getting the images from the incident that is passed in and putting them into the database

Reply:
Oh okay, but in reference to our other conversation, there may not be an image. Just make sure null is okay.